plugins/typescript/src/templates/context.ts

```diff
@@ -44,27 +44,28 @@ export const getContext = (prefix: string, componentsFile: string) =>
     return {
       fetcherOptions: {},
       queryOptions: {},
-      queryKeyFn: (operation) => {
-        const queryKey: unknown[] = hasPathParams(operation)
-          ? operation.path
-              .split("/")
-              .filter(Boolean)
-              .map((i) => resolvePathParam(i, operation.variables.pathParams))
-          : operation.path.split("/").filter(Boolean);
+      queryKeyFn: queryKeyFn
+  }
+};
 
-        if (hasQueryParams(operation)) {
-          queryKey.push(operation.variables.queryParams);
-        }
-
-        if (hasBody(operation)) {
-          queryKey.push(operation.variables.body);
-        }
-
-        return queryKey;
-      }
+  export const queryKeyFn = (operation:any) => {
```

Comment:
You can use `QueryOperation` from `{namespace}Components` type instead of `any`

Reply:
This might work but for me somehow some of my operations are missing then. Maybe it's because of my swagger input? I am not sure.

Reply:
i investigated more and found out that all the `update` operation ids are missing. So for all the `put` requests the operation types are not there.

Reply:
my bad, i had imported the wrong `QueryOperation`
so i guess having them in a separate file would be the best solution. otherwise they could be inconsistent.

Reply:
So far we are just generating keys for queries, since this is what data you are caching. I was actually not aware that mutations had keys, I will have a look on the doc to see the usage of those 🤔

Reply:
Refs:
https://react-query-v3.tanstack.com/guides/query-keys

![image](https://user-images.githubusercontent.com/1761469/207899720-4e02455c-9cc7-4c4d-8fc1-e2e32ff361f4.png)

So yeah, `mutationKey` is just for debugging, and you can still add them manually if needed 😉

Reply:
Ok for the queryFunctions generated i need all the operations, otherwise it will not work out of the box when the type is assigned to `Operation`. or am i getting something wrong here? i mean, yeah it's optional but 😸

Reply:
I get why you want a key, but I'm questioning the fact that we need to generate queryFunction for mutationssince this is to use with `queryClient.fetchQuery`